tasktiger/tasktiger.py

```diff
@@ -466,6 +475,92 @@ def get_queue_stats(self):
 
         return queue_stats
 
+    def purge_errored_tasks(
+        self,
+        queues=None,
+        exclude_queues=None,
+        last_execution_before=None,
+        limit=5000,
+    ):
+        """Purge failed tasks left in the ERROR state
+
+        Example usage::
+
+            limit = 500
+            n_processed = limit
+            while n_processed == limit:
+                n_processed = tiger.purge_errored_tasks(
+                    queues=['my-queue'],
+                    exclude_queues=['other-queue'],
+                    last_execution_before=(
+                        datetime.datetime.utcnow()
+                        - datetime.timedelta(days=14)
+                    ),
+                    limit=limit,
+                ):
+                time.sleep(.1)  # don't overload redis
+
+            # or to purge all in one go
+            tiger.purge_errored_tasks(limit=None)
+
+        :param iterable(str) queues: Queues to include
+        :param iterable(str) exclude_queues: Queues to exclude
+        :param datetime last_execution_before: If provided, only deletes tasks
+            older than this.
+        :param int limit: Stops after purging ``limit`` tasks. Can be
+            explicitly set to None to remove limit and purge all at once.
+
+        :returns: The total number of tasks purged
+        """
+        # some sanity checking and kind error messages for arguments
+        if last_execution_before:
+            assert isinstance(last_execution_before, datetime.datetime)
+        if limit is not None:
+            assert limit > 0, 'If specified, limit must be greater than zero'
+
+        only_queues = set(queues or self.config['ONLY_QUEUES'] or [])
+        exclude_queues = set(
+            exclude_queues or self.config['EXCLUDE_QUEUES'] or []
+        )
+
+        def errored_tasks():
+            queues_with_errors = self.connection.smembers(self._key(ERROR))
+            for queue in queues_with_errors:
+                if not queue_matches(
+                    queue,
+                    only_queues=only_queues,
+                    exclude_queues=exclude_queues,
+                ):
+                    continue
+
+                skip = 0
+                total = None
+                task_limit = limit or 1000
+                while total is None or skip < total:
+                    total_tasks, tasks = Task.tasks_from_queue(
+                        self, queue, ERROR, skip=skip, limit=task_limit
+                    )
+                    for task in tasks:
+                        if (
+                            last_execution_before
+                            and task.ts
+                            and task.ts > last_execution_before
+                        ):
+                            continue
+                        yield task
+                    skip += task_limit
+                    total = total_tasks
```

Comment:
I haven't tested this to confirm but it looks like the overall limit is being applied to the number of tasks we look at versus the number of tasks we delete. For instance, if we had 1001 total errored tasks and the first 1000's `task.ts` was greater than `last_execution_before` but the 1001 task wasn't, I don't think we would delete the 1001 task if `purge_errored_tasks()` was called with `limit=1000`.